File: lib/keeper/keeper_activation_mode.ml
type t = Manual | On_demand | Autonomous
let default = Autonomous
let restore_owner = function Manual -> false | On_demand | Autonomous -> true
let spontaneous = function Manual | On_demand -> false | Autonomous -> true
let to_string = function Manual -> "manual" | On_demand -> "on_demand" | Autonomous -> "autonomous"
let of_string = function
  | "manual" -> Some Manual
  | "on_demand" -> Some On_demand
  | "autonomous" -> Some Autonomous
  | _ -> None
let to_yojson mode = `String (to_string mode)

File: dashboard/src/lib/keeper-activation-mode.ts
export const KEEPER_ACTIVATION_MODES = ['manual', 'on_demand', 'autonomous'] as const
export type KeeperActivationMode = typeof KEEPER_ACTIVATION_MODES[number]

export function parseKeeperActivationMode(value: unknown): KeeperActivationMode | null {
  switch (value) {
    case 'manual': case 'on_demand': case 'autonomous': return value
    default: return null
  }
}

export function requireKeeperActivationMode(value: unknown): KeeperActivationMode {
  const mode = parseKeeperActivationMode(value)
  if (mode === null) throw new Error('Invalid keeper activation_mode')
  return mode
}

export const KEEPER_ACTIVATION_LABELS: Record<KeeperActivationMode, string> = {
  manual: '수동 시작',
  on_demand: '요청 시 실행',
  autonomous: '자율 실행',
}
